components/tooltip/index.tsx

```diff
@@ -77,17 +77,23 @@ export default class Tooltip extends React.Component<TooltipProps, any> {
 
   componentWillReceiveProps(nextProps: TooltipProps) {
     if ('visible' in nextProps) {
-      this.setState({ visible: nextProps.visible });
+      let onVisibleChange = this.props.onVisibleChange;
```

Comment:
Shouldn't this be const onVisibleChange. Also it will better if you use object destructuring as it is being used at other places as well. const {onVisibleChange} = this.props